File: programovani/ulohy/files/nahodne-zamichani-pole/fisher-yates-shuffle.js
// Deklarace a inicializace
const vel = 7; // Inicializujeme velikost pole
var pole = []; // Deklarujeme pole

// Cykls pro nauplneni pole cisly
for(var i = 0; i < vel; i++) {

    // Na index vlozime hodnotu
    pole[i] = i * 10;

    // Vypiseme ve tvaru klic - hodnota
    console.log(i + " - " + pole[i]);

}

// Vypiseme oddelovac
console.log("-----");

// Cyklus pro randomizaci
for (var i = vel - 1; i > 0 ;i--) {

    // Deklarujeme a inicializujeme
    var j = Math.floor(Math.random() * (i + 1)); // Inicializujeme, ziskame nahodnou pozici

    // Prohodime hodnoty
    [pole[j], pole[i]] = [pole[i], pole[j]];

}


// Cyklus pro vypsani pole
for(var i = 0; i < vel; i++) {

    // Vypiseme ve tvaru klic - hodnota
    console.log(i + " - " + pole[i]);

}

File: programovani/ulohy/files/nahodne-zamichani-pole/fisher-yates-shuffle.py
# Importujeme knihovnu pseudonahodnych cisel
import random

# Deklarace a inicializace
vel = 7 # Inicializujeme velikost pole
pole = [] # Deklarujeme pole

# Cyklus pro naplneni pole cisly
for i in range(vel):

    # Na index vlozime hodnotu
    pole.append(i * 10)

    # Vypiseme ve tvaru klic - hodnota
    print(i, "-", pole[i])

# Vypiseme oddelovac
print("-----")

# Cyklus pro randomizaci
for i in range(vel-1, -1, -1):

    # Deklarujeme a inicializujeme
    j = random.randrange(i + 1) # Inicializujeme, ziskame nahodnou pozici

    # Prohodime hodnoty
    pole[i], pole[j] = pole[j], pole[i]

# Cyklus pro vypsani pole
for i in range(vel):

    # Vypiseme ve tvaru klic - hodnota
    print(i, "-", pole[i])